ganttproject/src/biz/ganttproject/impex/csv/GanttCSVExport.java

```diff
@@ -102,25 +97,84 @@ public void save(OutputStream stream) throws IOException {
       format = format.withQuote(csvOptions.sSeparatedTextChar.charAt(0));
     }
 
-    CSVPrinter csvPrinter = new CSVPrinter(writer, format);
+    return format;
+  }
+
+  private SpreadsheetWriter getWriter(OutputStream stream, String formatName) throws IOException {
+    if (formatName != null && formatName.length() > 0) {
```

Comment:
I will add enum, but almost nothing will change. Firstly, if the enum is expanded, and in the verification does not add a new extension, the case default will work, and secondly, it can be null.